core/serialization/exceptions.js

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Contains custom errors thrown by the serialization system.
+ */
+'use strict';
+
+goog.module('Blockly.serialization.exceptions');
+goog.module.declareLegacyNamespace();
+
+
+// eslint-disable-next-line no-unused-vars
+const {State} = goog.requireType('Blockly.serialization.blocks');
+
+class DeserializationError extends Error { }
+exports.DeserializationError = DeserializationError;
+
+/**
+ * Represents an error where the serialized state is expected to provide a
+ * block type, but it is not provided.
+ */
+class MissingBlockType extends DeserializationError {
+  /**
+   * @param {!State} state The state object which is missing the block type.
+   */
+  constructor(state) {
+    super(`Expected to find a 'type' property, defining the block type`);
+
+    /**
+     * The state object containing the bad name.
+     * @type{!State}
```

Comment:
The rest of the codebase appears to have a space after '@type'.  Six instances in this PR.